Makefile

```diff
@@ -56,23 +64,33 @@ tag_latest:
 	docker tag $(NAME)/node-firefox:$(VERSION) $(NAME)/node-firefox:latest
 	docker tag $(NAME)/node-phantomjs:$(VERSION) $(NAME)/node-phantomjs:latest
 
-release: tag_latest
-	@if ! docker images $(NAME)/base | awk '{ print $$2 }' | grep -q -F $(VERSION); then echo "$(NAME)/base version $(VERSION) is not yet built. Please run 'make build'"; false; fi
-	@if ! docker images $(NAME)/hub | awk '{ print $$2 }' | grep -q -F $(VERSION); then echo "$(NAME)/hub version $(VERSION) is not yet built. Please run 'make build'"; false; fi
-	@if ! docker images $(NAME)/node-base | awk '{ print $$2 }' | grep -q -F $(VERSION); then echo "$(NAME)/node-base version $(VERSION) is not yet built. Please run 'make build'"; false; fi
-	@if ! docker images $(NAME)/node-chrome | awk '{ print $$2 }' | grep -q -F $(VERSION); then echo "$(NAME)/node-chrome version $(VERSION) is not yet built. Please run 'make build'"; false; fi
-	@if ! docker images $(NAME)/node-firefox | awk '{ print $$2 }' | grep -q -F $(VERSION); then echo "$(NAME)/node-firefox version $(VERSION) is not yet built. Please run 'make build'"; false; fi
-	@if ! docker images $(NAME)/node-phantomjs | awk '{ print $$2 }' | grep -q -F $(VERSION); then echo "$(NAME)/node-phantomjs version $(VERSION) is not yet built. Please run 'make build'"; false; fi
-	docker push $(NAME)/base
-	docker push $(NAME)/hub
-	docker push $(NAME)/node-base
-	docker push $(NAME)/node-chrome
-	docker push $(NAME)/node-firefox
-	docker push $(NAME)/node-phantomjs
+check_release_version:
+	@if [[ ! $(VERSION) =~ [[:digit:]].[[:digit:]].[[:digit:]] ]]; then echo "'$(VERSION)' is not an acceptable revision for a release target."; false; fi
+
+release: check_release_version tag_latest deploy
 	@echo "*** Don't forget to create a tag. git tag v$(VERSION) && git push origin v$(VERSION)"
 
+check_dev_release_version:
+		@if [[ $(VERSION) =~ [[:digit:]].[[:digit:]].[[:digit:]] ]]; then echo "'$(VERSION)' is not an acceptable revision for a develop target."; false; fi
```

Comment:
After more digging, I'm not sure there is a convention here. I see 'official' projects on the docker hub using latest for that last stable release. I see other projects using latest for their recent build. Furthermore, I see projects only ever using latest.  I believe documentation will have to convey our approach. I'm proposing;

Released revision: `major`.`minor`.`fix`-`[qualifier]`
Most recent CI build: `develop`

- `qualifier` can be something like `alpha`, `beta`, a release code name, etc -- but shouldn't be `snapshot`
- Only a released revision can be tagged `latest`, when it is warranted.

@sebady - How does this sound to you? Are you aware of any additional information that suggests we do something different?